libkpod/create.go

```diff
@@ -0,0 +1,823 @@
+		specgen.SetLinuxResourcesPidsLimit(c.config.PidsLimit)
+	}
+
+	// by default, the root path is an empty string. set it now.
+	specgen.SetRootPath(mountPoint)
+
+	saveOptions := generate.ExportOptions{}
+	if err = specgen.SaveToFile(filepath.Join(containerInfo.Dir, "config.json"), saveOptions); err != nil {
+		return nil, err
+	}
+	if err = specgen.SaveToFile(filepath.Join(containerInfo.RunDir, "config.json"), saveOptions); err != nil {
+		return nil, err
+	}
+
+	container, err := oci.NewContainer(containerID, containerName, containerInfo.RunDir, logPath, sb.NetNs(), labels, kubeAnnotations, image, metadata, sb.ID(), containerConfig.Tty, containerConfig.Stdin, containerConfig.StdinOnce, sb.Privileged(), sb.Trusted(), containerInfo.Dir, created, containerImageConfig.Config.StopSignal)
+	if err != nil {
+		return nil, err
+	}
+
+	return container, nil
+}
+
+func (c *ContainerServer) setupSeccomp(specgen *generate.Generator, cname string, sbAnnotations map[string]string) error {
+	profile, ok := sbAnnotations["security.alpha.kubernetes.io/seccomp/container/"+cname]
+	if !ok {
+		profile, ok = sbAnnotations["security.alpha.kubernetes.io/seccomp/pod"]
+		if !ok {
+			// running w/o seccomp, aka unconfined
+			profile = seccompUnconfined
+		}
+	}
+	if !c.seccompEnabled {
+		if profile != seccompUnconfined {
+			return fmt.Errorf("seccomp is not enabled in your kernel, cannot run with a profile")
+		}
+		logrus.Warn("seccomp is not enabled in your kernel, running container without profile")
+	}
+	if profile == seccompUnconfined {
+		// running w/o seccomp, aka unconfined
+		specgen.Spec().Linux.Seccomp = nil
+		return nil
+	}
+	if profile == seccompRuntimeDefault {
+		return seccomp.LoadProfileFromStruct(c.seccompProfile, specgen)
+	}
+	if !strings.HasPrefix(profile, seccompLocalhostPrefix) {
+		return fmt.Errorf("unknown seccomp profile option: %q", profile)
+	}
+	//file, err := ioutil.ReadFile(filepath.Join(s.seccompProfileRoot, strings.TrimPrefix(profile, seccompLocalhostPrefix)))
+	//if err != nil {
+	//return err
+	//}
+	// TODO(runcom): setup from provided node's seccomp profile
+	// can't do this yet, see https://issues.k8s.io/36997
+	return nil
+}
+
+// getAppArmorProfileName gets the profile name for the given container.
+func (c *ContainerServer) getAppArmorProfileName(annotations map[string]string, ctrName string) string {
+	profile := apparmor.GetProfileNameFromPodAnnotations(annotations, ctrName)
+
+	if profile == "" {
+		return ""
+	}
+
+	if profile == apparmor.ProfileRuntimeDefault {
+		// If the value is runtime/default, then return default profile.
+		return c.appArmorProfile
+	}
+
+	return strings.TrimPrefix(profile, apparmor.ProfileNamePrefix)
+}
+
+// openContainerFile opens a file inside a container rootfs safely
+func openContainerFile(rootfs string, path string) (io.ReadCloser, error) {
+	fp, err := symlink.FollowSymlinkInScope(filepath.Join(rootfs, path), rootfs)
+	if err != nil {
+		return nil, err
+	}
+	return os.Open(fp)
```

Comment:
Wrap this in a err := os.Open(fp) 



errors.Wrapf(err, "Unable to open %q", path )